src/web/source.rs

```diff
@@ -217,24 +222,45 @@ pub fn source_browser_handler(req: &mut Request) -> IronResult<Response> {
     };
 
     let storage = extension!(req, Storage);
-    let config = extension!(req, Config);
+    let archive_storage: bool = {
+        let rows = ctry!(
+            req,
+            conn.query(
+                "
+                SELECT archive_storage
+                FROM releases 
+                INNER JOIN crates ON releases.crate_id = crates.id
+                WHERE 
+                    name = $1 AND 
+                    version = $2
+                ",
+                &[&crate_name, &version]
+            )
+        );
+
+        let row = rows.get(0).ok_or(Nope::VersionNotFound)?;
```

Comment:
This looks wrong - we already checked this version exists above, in the call to `match_version`. I think this should just use `?` without changing the error.
```suggestion
        let row = rows.get(0)?;
```

Reply:
`rows.get` returns an `Option`, `ok_or` changes this to a `Result` to be used with `?`. 

Is there another / nicer way to achieve the result? `.unwrap` since we're sure the version exists?

Reply:
unwrap() seems good - it really should never be hit, and it will properly return a 500 if it is, rather than giving wrong info.